exir/tests/test_memory_format_ops_pass.py

```diff
@@ -124,8 +125,33 @@ def test_op_dim_order_propagation(self) -> None:
                 target_memory_format=torch.channels_last,
                 _load_for_executorch_from_buffer=_load_for_executorch_from_buffer,
             ),
+            check_unambiguous_dim_order=True,
         )
 
+    def test_op_dim_order_propagation_ambiguous(self) -> None:
+        try:
+            MemoryFormatOpsPassTestUtils.memory_format_test_runner(
+                self,
+                MemoryFormatTestSet(
+                    module=PropagateToCopyChannalsLastModule().eval(),
+                    op=torch.ops.aten._to_copy.default,
+                    sample_input=(
+                        torch.rand_like(
+                            torch.zeros(
+                                [2, 1, 2, 2]
+                            ),  # Ambiguous shape should trigger AmbiguousDimOrderError!
+                            dtype=torch.float32,
+                            memory_format=torch.contiguous_format,
+                        ),
+                    ),
+                    target_memory_format=torch.channels_last,
+                    _load_for_executorch_from_buffer=_load_for_executorch_from_buffer,
+                ),
+                check_unambiguous_dim_order=True,
+            )
+        except AmbiguousDimOrderError as e:
```

Comment:
the test right now can be passed when
a. AmbiguousDimOrderError raised, or
b. no error occurs.

b is not what we expected. 
I would like to add an assertion in the bottom of try section,
```suggestion
        assert False, "This test should be failed"
        except AmbiguousDimOrderError as e:
```